### PR TITLE
Final Project Submission (Movies Hub)

### PR DESCRIPTION
**Pull request by edenlisk**
We created the project with following features:
1. Homepage containing cards with data from TVamazeapi
2. Contact containing additional information
3. Receive and send likes and comments to the involvementAPI
4. Respective modal window for each card on the homepage

[Here is the link to the presentation video](https://youtu.be/5Kxg6o9zje4)

### CODE REVIEW

File: src/modules/counters.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const countItems = (container) => container.length;
```
Comment (gemmen29) [inline src/modules/counters.js]: - [x] Kindly make sure to follow the guidelines in the card `Add all items counter on the Homepage - Student A` in this [issues](https://github.com/edenlisk/movies-hub/issues/6). The `count-items` function needs to look for a specific dom element and make a count based on what is displayed as per this [requirement](https://github.com/edenlisk/movies-hub/issues/6#:~:text=A%20counter%20function%20should%20look%20for%20specific%20DOM%20elements%20(e.g.%20for%20the%20comments%20counter%20it%20should%20look%20for%20comments)%20and%20make%20the%20counting%20based%20on%20what%20is%20actually%20displayed%20on%20the%20page.).

File: src/modules/counters.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+
+export const countComments = (commentsContainer) => commentsContainer.length;
```
Comment (gemmen29) [inline src/modules/counters.js]: - [x] Kindly make sure to follow the guidelines in the card `Add comments counter - Student B` in this [issues](https://github.com/edenlisk/movies-hub/issues/5). The `countComments` function needs to look for a specific dom element and make a count based on what is displayed as per this [requirement](https://github.com/edenlisk/movies-hub/issues/5#:~:text=A%20counter%20function%20should%20look%20for%20specific%20DOM%20elements%20(e.g.%20for%20the%20comments%20counter%20it%20should%20look%20for%20comments)%20and%20make%20the%20counting%20based%20on%20what%20is%20actually%20displayed%20on%20the%20page.).

File: src/test/counters.test.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const { countItems } = require('../modules/counters.js');
+
+describe('Counting number of movies on the page', () => {
+  test('test number of items', () => {
```
Comment (gemmen29) [inline src/test/counters.test.js]: - [x] Kindly update the test for the `count-items` function based on the new implementation and please make sure that your test includes all edge cases like if there are 0 items on the page.

File: src/test/counters.test.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          original: 'https://static.tvmaze.com/uploads/images/original_untouched/82/206879.jpg',
+        },
+        title: 'Revenge',
+        language: 'English',
+        genres: ['Drama', 'Thriller', 'Mystery'],
+      },
+    ];
+    const result = countItems(cardsContainer);
+    expect(result).toEqual(3);
+  });
```
Comment (gemmen29) [inline src/test/counters.test.js:34]: - [x] Kindly include tests for the `countComments` function also based on the new implementation and please make sure that your test includes all edge cases like if there are 0 items on the page.

File: src/modules/liking.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import { countComments } from './counters.js';
+
```
Comment (gemmen29) [inline src/modules/liking.js:2]: - [ ] [OPTIONAL] Kindly extract the calling API logic in an external module instead of mixing dom code with API calling and use it here to make your code more cleaner and organized.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

**Hi @Team,** 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented suggestions.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "Fixing bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```
